Indexer arm64 package generation failing due to `aws: command not found`

## Description

While analyzing https://github.com/wazuh/wazuh-qa-automation/issues/5854, it has been detected that the indexer arm64 package generation workflows are failing due to:

```nim
/home/runner/work/_temp/eb7efca6-abeb-475b-aa2c-008f6e74a33c.sh: line 3: aws: command not found
Error: Process completed with exit code 127.
```

## Evidences

- indexer-deb-arm64 -> https://github.com/wazuh/wazuh-indexer/actions/runs/22980360157/job/66719735282
- indexer-rpm-aarch64 -> https://github.com/wazuh/wazuh-indexer/actions/runs/22980361196/job/66719752539

--- a/build-scripts/provision.sh
+++ b/build-scripts/provision.sh
@@ -33,7 +33,7 @@ if [[ "$ARCH" == "aarch64" ]]; then
     sudo apt-get update
     sudo apt-get install -y maven rpm cpio
     sudo apt-get update && sudo apt-get install -y unzip
-    curl "https://awscli.amazonaws.com/awscli-exe-linux-x86_64.zip" -o "awscliv2.zip"
+    curl "https://awscli.amazonaws.com/awscli-exe-linux-aarch64.zip" -o "awscliv2.zip"
     unzip -q awscliv2.zip
     sudo ./aws/install --update
     

